stats/stats/src/charts/chart.rs

```diff
@@ -0,0 +1,72 @@
+use async_trait::async_trait;
+use entity::{charts, sea_orm_active_enums::ChartType};
+use sea_orm::{prelude::*, sea_query, FromQueryResult, QuerySelect, Set};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum UpdateError {
+    #[error("blockscout database error: {0}")]
+    BlockscoutDB(DbErr),
+    #[error("stats database error: {0}")]
+    StatsDB(DbErr),
+    #[error("chart {0} not found")]
+    NotFound(String),
+    #[error("internal error: {0}")]
+    Internal(String),
+}
+
+#[async_trait]
+pub trait Chart: Sync {
+    fn name(&self) -> &str;
+    fn chart_type(&self) -> ChartType;
+
+    async fn create(&self, db: &DatabaseConnection) -> Result<(), DbErr> {
+        create_chart(db, self.name().into(), self.chart_type()).await
+    }
+
+    async fn update(
+        &self,
+        db: &DatabaseConnection,
+        blockscout: &DatabaseConnection,
+        force_full: bool,
+    ) -> Result<(), UpdateError>;
+}
+
+#[derive(Debug, FromQueryResult)]
+struct ChartID {
+    id: i32,
+}
+
+pub async fn find_chart(db: &DatabaseConnection, name: &str) -> Result<Option<i32>, DbErr> {
```

Comment:
btw, can we make `find_chart()` is part of Chart trait with default impl, and call it `id()` for example? It seems for logical to me

Reply:
we can do that, but may be in another pr? don't see a reason to do it here